lib/auth/default_current_user_provider.rb

```diff
@@ -128,33 +141,43 @@ def current_user
     # possible we have an api call, impersonate
     if api_key
       current_user = lookup_api_user(api_key, request)
-      raise Discourse::InvalidAccess.new(I18n.t('invalid_api_credentials'), nil, custom_message: "invalid_api_credentials") unless current_user
+      if !current_user
+        raise Discourse::InvalidAccess.new(
+          I18n.t('invalid_api_credentials'),
+          nil,
+          custom_message: "invalid_api_credentials"
+        )
+      end
       raise Discourse::InvalidAccess if current_user.suspended? || !current_user.active
+      admin_api_key_limiter.performed! if !Rails.env.profile?
       @env[API_KEY_ENV] = true
-      rate_limit_admin_api_requests!
     end
 
     # user api key handling
     if user_api_key
-
       hashed_user_api_key = ApiKey.hash_key(user_api_key)
-      limiter_min = RateLimiter.new(nil, "user_api_min_#{hashed_user_api_key}", GlobalSetting.max_user_api_reqs_per_minute, 60)
-      limiter_day = RateLimiter.new(nil, "user_api_day_#{hashed_user_api_key}", GlobalSetting.max_user_api_reqs_per_day, 86400)
+      @env[HASHED_USER_API_KEY] = hashed_user_api_key
```

Comment:
```suggestion
      @env[HASHED_USER_API_KEY] = hashed_user_api_key = ApiKey.hash_key(user_api_key)
```

Reply:
I'm also curious why we have to set the hashed user api key on the `@env`. It seems like it is currently being used by the rate limiter methods but I we can just pass the hashed user api key as a parameter to those methods.

Reply:
I've gone with storing the hashed key in a instance variable instead of `@env` and the rate limiters methods will just use the instance variable.